test/github-package.test.js

```diff
@@ -683,64 +910,183 @@ describe('GithubPackage', function() {
+      });
 
-      assert.isTrue(githubPackage.getActiveRepository().isPresent());
-      assert.strictEqual(
-        githubPackage.getActiveRepository(),
-        await contextPool.getContext(nonRepositoryPath).getRepository(),
-      );
+      contextPool = githubPackage.getContextPool();
+    });
+
+    afterEach(async function() {
+      await githubPackage.deactivate();
+
+      atomEnv.destroy();
+    });
+
+    context('with a non-repository project', function() {
+      let nonRepositoryPath;
+      beforeEach(async function() {
+        nonRepositoryPath = await getTempDir();
+        project.setPaths([nonRepositoryPath]);
+
+        await contextUpdateAfter(githubPackage, () => githubPackage.activate());
+        await githubPackage.getActiveRepository().getLoadPromise();
+
+        await githubPackage.initialize(nonRepositoryPath);
+      });
+
+      it('creates a repository for the project', function() {
+        assert.isTrue(githubPackage.getActiveRepository().isPresent());
+      });
+
+      it('uses the newly created repository for the project', async function() {
+        assert.strictEqual(
+          githubPackage.getActiveRepository(),
+          await contextPool.getContext(nonRepositoryPath).getRepository(),
+        );
+      });
     });
   });
 
-  describe('clone', function() {
-    it('clones into an existing project path', async function() {
-      const sourcePath = await cloneRepository();
-      const existingPath = await getTempDir();
-      project.setPaths([existingPath]);
+  describe('clone()', function() {
+    let atomEnv, githubPackage;
+    let workspace, project, commands, notificationManager;
+    let tooltips, deserializers, config, keymaps, styles;
+    let grammars, confirm, configDirPath, getLoadSettings;
+    let renderFn, contextPool, currentWindow;
+    let useLegacyPanels;
 
-      await contextUpdateAfter(() => githubPackage.activate());
-      const repository = githubPackage.getActiveRepository();
-      await repository.getLoadPromise();
-      assert.isTrue(repository.isEmpty());
+    beforeEach(async function() {
+      atomEnv = global.buildAtomEnvironment();
+      await disableFilesystemWatchers(atomEnv);
+
+      workspace = atomEnv.workspace;
+      project = atomEnv.project;
+      commands = atomEnv.commands;
+      deserializers = atomEnv.deserializers;
+      notificationManager = atomEnv.notifications;
+      tooltips = atomEnv.tooltips;
+      config = atomEnv.config;
+      keymaps = atomEnv.keymaps;
+      confirm = atomEnv.confirm.bind(atomEnv);
+      styles = atomEnv.styles;
+      grammars = atomEnv.grammars;
+      getLoadSettings = atomEnv.getLoadSettings.bind(atomEnv);
+      currentWindow = atomEnv.getCurrentWindow();
+      configDirPath = path.join(__dirname, 'fixtures', 'atomenv-config');
+      renderFn = sinon.stub().callsFake((component, element, callback) => {
+        if (callback) {
+          process.nextTick(callback);
+        }
+      });
```

Comment:
🤔 I'd love to find a way to avoid duplicating all of this. A lot of it might go away if we can move to hooks and use an Atom environment context, though, so I'm unsure if it would be worth the effort to refactor out now.

Reply:
I simplified it a little more, I think it looks nicer now.